pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AntlrBaseNode.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.ast;
+
+import java.util.Iterator;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+
+import net.sourceforge.pmd.lang.ast.xpath.Attribute;
+import net.sourceforge.pmd.lang.ast.xpath.AttributeAxisIterator;
+import net.sourceforge.pmd.lang.dfa.DataFlowNode;
+
+public class AntlrBaseNode extends ParserRuleContext implements AntlrNode {
```

Comment:
Since this class is a base implementation, maybe it should be abstract?

Reply:
We thought that at first too, but we left it plain class since we don't have abstract methods required yet, we may found some along the way while we integrate Antlr

Reply:
Also.. take a look at the branch name LOL 🤣 
<img width="248" alt="captura de pantalla 2019-02-17 a la s 17 22 47" src="https://user-images.githubusercontent.com/7543268/52918693-b014f900-32d8-11e9-8368-43eeaacea9d7.png">

Reply:
Yeah it's ironic! I would probably make it abstract anyway so as not to forget it later on. It's easier to remove an abstract modifier than to add one.

Reply:
I made a sticky note, otherwise [PMD will warn me](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#abstractclasswithoutabstractmethod) for it.

Given that we are going to extend `ParserRuleContext` it's possible to end up without needing an abstract class, because we have references to children, parent, etc. there, but we are not sure yet, it will be clearer along the way, specially when we get to the rules part